### ISSUE TITLE
Every session comes from the Docker host IP when using reverse proxy

### ISSUE THREAD

**Issue by martinvonwittich** [zammad/zammad-docker-compose#131]
### Infos:

* Docker version: `docker-ce 5:18.09.5~3-0~debian-buster`
* Docker-compose version: `docker-compose 1.21.0-3`
* Operating system (Docker host): Debian 10.3

### Expected behavior:

* Sessions in System -> Sessions should show the correct IPs.

### Actual behavior:

* All Session IPs are 172.17.0.1 (the Docker host IP of my server).

### Steps to reproduce the behavior:

* Run zammad-docker-compose
* Set up a reverse proxy (in my case Apache) that sits between zammad-docker-compose and the clients
* Login into the web interface, and open the Sessions list.

### Background information

This is related to https://github.com/zammad/zammad/issues/742, but the fix doesn't work in this case because the nginx container will add the Docker host IP to the `X-Forwarded-For` header. This is what the request looks like when it arrives at the nginx container:

```
    X-Forwarded-SSL: on
    X-Forwarded-For: 2003:a:422:...
    X-Forwarded-Host: zammad-test.ourcompany.tld
    X-Forwarded-Server: zammad-test.ourcompany.tld
```

This is what the request looks like when it arrives at the railsserver container:

```
    X-Forwarded-For: 2003:a:422:..., 172.17.0.1
    X-Forwarded-Proto: http
    X-Forwarded-SSL: on
    X-Forwarded-Host: zammad-test.ourcompany.tld
    X-Forwarded-Server: zammad-test.ourcompany.tld
```

I manually inserted the host's IP into my `config/environments/production.rb`, and this seems to resolve the issue:

```
zammad-test ~ # grep trusted /var/lib/docker/volumes/zammaddockercompose_zammad-data/_data/config/environments/production.rb
  config.action_dispatch.trusted_proxies = ['127.0.0.1', '::1', '172.17.0.1']
```

**Comment by MrGeneration** [comment]
@monotek can you please have a look when you have some spare time? I'll gladly assist to see if we want to have it in the repo directly or rather as extension of the documentation on how to solve this issue in a proper way without changing code? :)

**Comment by monotek** [comment]
My docker-compose network is 172.18.0.0/16 with the gateway @ 172.18.0.1.

I guess there is no genereal solution and we need to make this somehow configurable. 

Problem affects the zammad helm chart too.

**Comment by martinvonwittich** [comment]
> I guess there is no genereal solution and we need to make this somehow configurable.

Would determining and using the container's gateway IP (which should be host) an appropriate solution for this? Unfortunately this would probably require additional userspace tools like `route` or `ip` in the container; I fiddled a bit with it to figure out if this was easily queryable e.g. via `/proc` or `/sys`, but I haven't found a way yet.

**Comment by renatocron** [comment]
What I ended doing (solved #120 as well) was running the NGINX on the host (i was already running it to forward to the second nginx).

docker-compose.override.yml 
```
version: '2'
services:

  zammad-nginx:
    ports:
       - "172.17.0.1:8083:80"

  zammad-railsserver: 
    ports:
       - "172.17.0.1:8084:3000"
  zammad-websocket:
    ports:
       - "172.17.0.1:8082:6042"
```

then on nginx config:

```    upstream zammad {
        server 172.17.0.1:8084;
    }

    upstream zammadws {
        server 172.17.0.1:8082;
    }
```


For some reason I need to forward `/assets` to nginx container (http://172.17.0.1:8083) in order to some assets load (some fonts, chats mp3, form.js)

**Comment by monotek** [comment]
fixed by: https://github.com/zammad/zammad-docker-compose/pull/166
